airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/test/util/destination_process/DockerizedDestination.kt

```diff
@@ -196,13 +196,13 @@ class DockerizedDestination(
     override suspend fun shutdown() {
         withContext(Dispatchers.IO) {
             destinationStdin.close()
-            // The old cdk had a 1-minute timeout here. That seems... weird?
-            // We can just rely on the junit timeout, presumably?
-            process.waitFor()
             // Wait for ourselves to drain stdout/stderr. Otherwise we won't capture
             // all the destination output (logs/trace messages).
             stdoutDrained.join()
             stderrDrained.join()
+            // The old cdk had a 1-minute timeout here. That seems... weird?
+            // We can just rely on the junit timeout, presumably?
+            process.waitFor()
```

Comment:
Does Junit have a default timeout? https://junit.org/junit5/docs/5.5.1/api/org/junit/jupiter/api/Timeout.html?

`runTest` timeout after 60 seconds but we're not using that here, right?

Reply:
I was actually talking to stephane about this recently - there's some magic in [LoggingInvocationInterceptor](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/bulk/core/base/src/testFixtures/kotlin/io/airbyte/cdk/extensions/LoggingInvocationInterceptor.kt) to enforce the junit timeout. Didn't look deeply into it though (and haven't actually tested if it plays well with coroutines + docker), that's on the todo list

(right below "figure out why junit isn't executing tests concurrently" >.>)